pkgs/tools/misc/hdf5/default.nix

```diff
@@ -8,14 +8,18 @@
 , enableShared ? true
 }:
 
-with { inherit (stdenv.lib) optional; };
+# cpp and mpi opitons are mutually exclusive
+# (--enable-unsupported could be used to force the build)
+assert cpp -> mpi == null;
```

Comment:
I just got to think about it in a symmetrical way, and implications feel rather asymmetric to me.